src/components/Categories/Categories.jsx

```diff
@@ -1,7 +1,31 @@
+import { useDispatch, useSelector } from 'react-redux';
+import productSlice from '../../store/products';
+
 export default function Categories() {
+  const dispatch = useDispatch();
+  const categories = useSelector((state) => state.product.categories);
```

Comment:
try changing this part `tate.product.categories` to this `state.product.productData.categories`